Content.Server/Body/Systems/RespiratorSystem.cs

```diff
@@ -109,26 +114,94 @@ public override void Update(float frameTime)
         }
     }
 
-    public void Inhale(EntityUid uid, BodyComponent? body = null)
+    public void Inhale(Entity<RespiratorComponent?, BodyComponent?, ThermalRegulatorComponent?> ent)
     {
-        if (!Resolve(uid, ref body, logMissing: false))
+        if (!Resolve(ent, ref ent.Comp1, ref ent.Comp2, logMissing: false))
             return;
 
-        var organs = _bodySystem.GetBodyOrganEntityComps<LungComponent>((uid, body));
+        var organs = _bodySystem.GetBodyOrganEntityComps<LungComponent>((ent, ent.Comp2));
 
         // Inhale gas
         var ev = new InhaleLocationEvent();
-        RaiseLocalEvent(uid, ref ev);
+        RaiseLocalEvent(ent, ref ev);
 
-        ev.Gas ??= _atmosSys.GetContainingMixture(uid, excite: true);
+        ev.Gas ??= _atmosSys.GetContainingMixture(ent.Owner, excite: true);
 
         if (ev.Gas is null)
         {
             return;
         }
 
+        // Remove a volume of gas from the environment
         var actualGas = ev.Gas.RemoveVolume(Atmospherics.BreathVolume);
 
+        // Only apply temperature damage if there are actually moles to breathe
+        // This prevents damage from vacuum which has temperature but no actual gas
+        if (actualGas.TotalMoles > Atmospherics.GasMinMoles)
+        {
+            var gasTemp = actualGas.Temperature;
+
+            // Take the normal body temperature from the mob's ThermalRegulatorComponent
+            // We don't resolve this component because it shouldn't be required, so we fall back to the atmos consts
+            var normalBodyTempK = ent.Comp3?.NormalBodyTemperature ?? Atmospherics.T0C + Atmospherics.NormalBodyTemperature;
+            var heatCapacity = _atmosSys.GetHeatCapacity(actualGas, false);
+
+            // Calculate thermal energy difference between inhaled gas and normal body temperature
+            // This represents how much thermal energy would be transferred to/from the body
+            var normalThermalEnergy = heatCapacity * normalBodyTempK;
+            var gasThermalEnergy = _atmosSys.GetThermalEnergy(actualGas);
+            var energyDifference = Math.Abs(gasThermalEnergy - normalThermalEnergy);
+
+            // Constants for damage scaling
+            // Oxygen at -200°C (73.15K) with its heat capacity will cause maximum damage (5x multiplier)
+            const float baseDamageEnergyThreshold = 25f;
+
+            // Cap damage multiplier at 5x to prevent extremely hot/cold small amounts of gas from dealing excessive damage
+            const float maxDamageMultiplier = 5.0f;
+
+            // Calculate a damage scale factor based on thermal energy difference
+            // The factor starts at 1.0 at the base threshold and increases linearly with energy
+            var energyScaleFactor = 1.0f + (energyDifference / baseDamageEnergyThreshold);
+            energyScaleFactor = Math.Min(energyScaleFactor, maxDamageMultiplier);
```

Comment:
Consider using some combination of `Clamp` and `Lerp` to make this more readable. Or, put the whole equation on one line so someone can easily understand what the formula is and plot it.